### PR TITLE
[Core][User] Add administrator role

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Related tickets | #5797 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Rbac\Model\RoleInterface;
+use Sylius\Component\Rbac\Repository\RoleRepositoryInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class AddAdministratorRoleListener
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php]: final?

File: src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public function addAdministrationRole(GenericEvent $event)
+    {
+        /** @var AdminUserInterface $adminUser */
+        $adminUser = $event->getSubject();
+
+        /** @var RoleInterface $administratorRole */
+        $administratorRole = $this->roleRepository->findOneBy(['code' => 'administrator']);
+
+        if (null === $administratorRole) {
+            throw new \RuntimeException('Cannot add administration role, because cannot find administrator role');
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php]: Administrator role could not be found?

File: src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+        if (!$adminUser instanceof AdminUserInterface) {
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php]: This check should go before finding the role.

File: src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            throw new \RuntimeException('Cannot add administration role, because given subject is not admin user');
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/EventListener/AddAdministratorRoleListener.php]: Expected AdminUserInterface, got \some\other\class?

File: src/Sylius/Bundle/CoreBundle/Resources/config/services.xml [hunk 5]
```diff
@@ -235,6 +235,10 @@
         </service>
 
         <!-- listeners -->
+        <service id="sylius.listener.add_administrator_role" class="Sylius\Bundle\CoreBundle\EventListener\AddAdministratorRoleListener">
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services.xml]: public="false"

Comment (Arminek) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services.xml]: Event listeners are lazy loaded.

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services.xml]: Yup, forgot about that.

**Comment by pjedrzejewski** [comment]
Thanks Arek!

**Pull request merged**